modules/avcodec/encode.c

```diff
@@ -485,44 +490,49 @@ int avcodec_encode(struct videnc_state *st, bool update,
 	}
 
 	ret = avcodec_send_frame(st->ctx, hw_frame ? hw_frame : pict);
-	if (ret < 0) {
+	if (ret < 0 && ret != AVERROR(EAGAIN)) {
 		err = EBADMSG;
 		goto out;
 	}
 
-	ret = avcodec_receive_packet(st->ctx, pkt);
-	if (ret < 0) {
-		err = 0;
-		goto out;
-	}
+	while (1)
```

Comment:
```suggestion
	while (ret >= 0)
```
If avcodec_send_frame is EAGAIN, the while loop should not entered and is more robust